tests/test_service.py

```diff
@@ -84,7 +86,30 @@ def __iter__(self):
     rp_service.collect_tests(mocked_session)
     rp_service.start_pytest_item(mocked_item)
 
-    expect(mocked_item_start.call_count == 1, 'One HTTP POST sent')
+    expect(mocked_item_start.call_count == 2, 'Two HTTP POST sent')
     code_ref = mocked_item_start.call_args[1]['code_ref']
     expect(code_ref == '/path/to/test - test_item')
     assert_expectations()
+
+
+@pytest.mark.parametrize('os_type', ('windows', 'unix'))
+def test_adding_item_to_hierarchy(mocked_item, os_type):
+    """ Test adding path of module to hierarchy on both OS
+
+    :param mocked_item:       a mocked test item
+    :param os_type:           type of operation system
+    """
+    parent_path = mocked_item.parent.fspath
+    if os_type == 'windows':
+        parent_path = parent_path.new(dirname='C:/path/to/item')
+    else:
+        parent_path = parent_path.new(dirname='/path/to/item')
+    rp_name = PyTestServiceClass._add_item_hier_parts_other(
+        item_parts=PyTestServiceClass._get_item_parts(mocked_item),
+        item=mocked_item,
+        item_type=Module,
+        hier_flag=False,
+        report_parts=[],
+        rp_name="")
+    expect(rp_name == 'module.py')
```

Comment:
Soft asserts should be used only if u have multiple checks within one test case.

Reply:
fixed